opm/simulators/wells/MultisegmentWell.hpp

```diff
@@ -60,13 +60,12 @@ namespace Opm
         // sometimes, they are similar, while sometimes, they can have very different forms.
 
         // TODO: the following system looks not rather flexible. Looking into all kinds of possibilities
-        // TODO: gas is always there? how about oil water case?
-        // Is it gas oil two phase case?
         static const bool gasoil = numEq == 2 && (Indices::compositionSwitchIdx >= 0);
```

Comment:
This part has been confusing to read. Let us see whether we can find some good way in doing it. Probably we can do through the `FluidSystem`.

Reply:
One suggestion is as follows, 
```c++
          static const bool has_water = ....;
          static const bool has_gas = ....;
          static const int GTotal = 0;
          static const int WFrac = has_water ? 1 : -1000;
          static const int GFrac = has_gas ? has_water + has_gas : - 1000;
          static const int SPres = has_water + has_gas + 1;
```

The good thing about this way is easier to understand. With `gasoil`, we use some negative logic to tell whether water is there, which is why the code can be confusing to understand or more difficult to modify. 

For `SPres`, it might not be good to relate it directly to `NumEq`, which is the number of the reservoir equations. Maybe it is better to associate with `NumWellEq`. 

You can also name `-1000` as some named variable to indicate that it is just for an invalid index. 

About how to get `has_water` and `has_gas`. At the beginning I was thinking about `FluidSystem`. But it has some problem in compiling, not sure why.  Maybe `Indices`? (`Indices::waterEnabled` and `Indices::gasEnabled`). You understand these much better than me.